lib/wallet/test/unit/Cardano/Write/Tx/BalanceSpec.hs

```diff
@@ -0,0 +1,1954 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE EmptyCase #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE OverloadedLabels #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE TypeApplications #-}
+{- HLINT ignore "Use null" -}
+{- HLINT ignore "Use camelCase" -}
+
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+-- TODO: https://cardanofoundation.atlassian.net/browse/ADP-2841
+{-# LANGUAGE CPP #-}
+#if __GLASGOW_HASKELL__ >= 902
+{-# OPTIONS_GHC -fno-warn-ambiguous-fields #-}
+#endif
+
+module Cardano.Write.Tx.BalanceSpec
+    ( spec
+
+    ----------------------------------------------------------------------------
+    -- TODO: ADP-3171
+    -- The exports in this section are currently needed by other tests related
+    -- to balanceTx that have not yet been relocated to this module. Once all
+    -- balanceTx-related tests have been relocated to this module, we should
+    -- remove these exports:
+    , dummyPolicyK
+    , mockPParamsForBalancing
+    , testTxLayer
```

Comment:
`testTxLayer` should also not be needed if we reimplement `totalOutput`.

Reply:
To avoid reimplementing you could instead duplicate the definition of `testTxLayer`. I think that's preferable than having this dependency between the modules for a long time.

Reply:
> testTxLayer should also not be needed if we reimplement totalOutput.

Fair enough! Though in this PR I've purposefully avoided re-implementing anything. (To keep things simple.)

> To avoid reimplementing you could instead duplicate the definition of testTxLayer. I think that's preferable than having this dependency between the modules for a long time.

Hopefully these exports should disappear very soon! (In fact, as soon as we migrate the other `balanceTx`-related tests to the new module.

Reply:
The problem with `testTxLayer` I saw is that they are also used by the signTx tests, which won't be moved immediately.

Reply:
So just: if "Hopefully these exports should disappear very soon!" fails, please consider taking alternative approaches to break the dependency between the two spec

Reply:
> So just: if "Hopefully these exports should disappear very soon!" fails, please consider taking alternative approaches to break the dependency between the two spec

In that case, we can definitely consider duplicating the definition of `testTxLayer` in a future PR.